Cython/Compiler/ModuleNode.py

```diff
@@ -208,7 +216,14 @@ def h_entries(entries, api=0, pxd=0):
             h_code.putln("/* It now returns a PyModuleDef instance instead of a PyModule instance. */")
             h_code.putln("")
             h_code.putln("#if PY_MAJOR_VERSION < 3")
-            h_code.putln("PyMODINIT_FUNC init%s(void);" % env.module_name)
+            try:
+                env.module_name.encode("ascii")
```

Comment:
I've added an `isascii` method to `EncodedString` and `BytesLiteral` instead (for Py2 only). I haven't rewritten the punycoding code since this is taken from PEP489 and it seems best to match that.